Mathlib/MeasureTheory/Integral/IntervalIntegral/DerivIntegrable.lean

```diff
@@ -0,0 +1,154 @@
+/-
+Copyright (c) 2025 Yizheng Zhu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yizheng Zhu
+-/
+import Mathlib.Analysis.BoundedVariation
+import Mathlib.MeasureTheory.Function.AbsolutelyContinuous
+import Mathlib.MeasureTheory.Integral.IntervalIntegral.Slope
+
+/-!
+# `f'` is interval integrable for certain classes of functions `f`
+
+This file proves that:
+* `MonotoneOn.intervalIntegrable_deriv` - If `f` is monotone on `a..b`, then `f'` is interval
+integrable on `a..b`.
+* `MonotoneOn.intervalIntegral_deriv_bound` - If `f` is monotone on `a..b`, then the integral of
+`f'` on `a..b` is in `uIcc 0 (f b - f a)`.
+* `BoundedVariationOn.intervalIntegrable_deriv` - If `f` has bounded variation on `a..b`,
+then `f'` is interval integrable on `a..b`.
+* `AbsolutelyContinuousOnInterval.intervalIntegrable_deriv` - If `f` is absolutely continuous on
+`a..b`, then `f'` is interval integrable on `a..b`.
+
+## Tags
+interval integrable, monotone, bounded variation, absolutely continuous
+-/
+
+open MeasureTheory Set Filter
+
+open scoped Topology
+
+/-- If `f` is monotone on `[a, b]`, then `f'` is the limit of `G n` a.e. on `[a, b]`, where each
+`G n` is `AEStronglyMeasurable` and the liminf of the lower Lebesgue integral of `‖G n ·‖ₑ` is at
+most `f b - f a`. -/
+lemma MonotoneOn.exists_tendsto_deriv_liminf_lintegral_enorm_le
+    {f : ℝ → ℝ} {a b : ℝ} (hab : a ≤ b) (hf : MonotoneOn f (Icc a b)) :
+    ∃ G : (ℕ → ℝ → ℝ), (∀ᵐ x ∂volume.restrict (Icc a b),
+      Filter.Tendsto (fun (n : ℕ) ↦ G n x) Filter.atTop (𝓝 (deriv f x))) ∧
+      (∀ (n : ℕ), AEStronglyMeasurable (G n) (volume.restrict (Icc a b))) ∧
+      liminf (fun (n : ℕ) ↦ ∫⁻ (x : ℝ) in Icc a b, ‖G n x‖ₑ) atTop ≤
+        ENNReal.ofReal (f b - f a) := by
+  let g (x : ℝ) : ℝ := f (max a (min x b))
```

Comment:
Could you add a sketch of the proof in a comment at the beginning of the proof? It's often useful whenever a proof is nontrivial, to help the reader follow what is going on (in particular, here, to understand why it's useful to extend the function to a globally monotone function)